### PR TITLE
chore(v3): core happy path v3

### PR DESCRIPTION
**Pull request by jayzhang**
Happy path for FxCore v3 with two plugins: ReactTabScaffoldPlugin and AzureStoragePlugin:
1. init
2. addModule
3. scaffold
4. addResource

### CODE REVIEW

**Review by LongOddCode — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/frontend/ops/deploy.ts [hunk 1]
```diff
@@ -126,6 +153,40 @@ export class FrontendDeployment {
...
   }
 
+  public static async doFrontendDeploymentV3(
+    client: AzureStorageClient,
+    componentPath: string,
+    envName: string,
+    progress?: IProgressHandler
+  ): Promise<void> {
+    const skip = await FrontendDeployment.needDeploy(componentPath, envName);
+    if (!skip) {
```
Comment (hund030) [inline packages/fx-core/src/plugins/resource/frontend/ops/deploy.ts:163]: Reverse condition.

File: packages/fx-core/src/plugins/resource/frontend/v3/ReactTabScaffoldPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            throw new UnknownScaffoldError();
+        }
+      },
+    });
+    await progress.end(true);
+    ctx.logProvider.info(Messages.EndScaffold(this.name));
+    const module = getModule(solutionSettings, inputs.module);
+    if (module) {
+      module.dir = FrontendPathInfo.WorkingDir;
+      module.deployType = "folder";
```
Comment (hund030) [inline packages/fx-core/src/plugins/resource/frontend/v3/ReactTabScaffoldPlugin.ts:105]: Can I add customize config of module? For React tab, I may need to add module.indexPath = "/index.html#"

**Review by nliu-ms — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/frontend/v3/ReactTabScaffoldPlugin.ts (hunk 2 above)
Comment (hund030) [reply packages/fx-core/src/plugins/resource/frontend/v3/ReactTabScaffoldPlugin.ts:105]: And I remember that scaffold plugin can contribute to teams app manifest template, how to achieve that?

**Review by XiaofuHuang — APPROVED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/debug/scaffolding.ts [hunk 3]
```diff
@@ -238,15 +244,15 @@ export async function _scaffoldLocalDebugSettings(
     TelemetryEventName.scaffoldLocalDebugSettings,
     telemetryProperties
   );
-  return ok(Void);
+  return ok(localSettings as Json);
```
Comment (kuojianlu) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/scaffolding.ts:247]: The declared return type of this function is `Promise<Result<Void, FxError>>`, why return `ok(localSettings as Json)` here?

**Review by hund030 — APPROVED** [review]
Skip build condition should be reversed. Other parts look good to me.

**Pull request merged**